libraries/helpers.rb

```diff
@@ -25,6 +25,7 @@ module Helpers
 
       def installed_postgresql_major_version
         pgsql_package = node['packages'].filter { |p| p.match?(/^postgresql-?(\d+)?$/) }
+        pgsql_package = pgsql_package.sort_by { |key, _values| key[1].to_i }.to_h
```

Comment:
This doesn't seem to work properly and I'm confused why you're converting that value to an int. If I just change it to the following, it seems to work properly.
```suggestion
        pgsql_package = pgsql_package.sort_by { |key, _values| key }.to_h
```

Reply:
I agree with @ramereth here, it doesn't work nor make sense from what I can work out. I'd sort against the `version` key if we wanted to always order by version but that doesn't necessarily help us out.

I'm thinking the better option would be to move the package match logic into a seperate method which `installed_postgresql_major_version` and `installed_postgresql_package_source` could use to get the package so they'll always tie up.

Further to that, I think that matching more than one package is an unknown that we can't handle 100% so it might be better to `raise` if we get more than one package matched.

Reply:
Really appreciate the suggestions on how to improve this. @bmhughes Moving the matching logic into a function I agree makes a lot of sense here. With that said, the goal of sorting was to select for the newest PG version in order to somewhat accommodate multiple PG installations. 

What are the thoughts on implementing a property overriding the match found from either `installed_postgresql_major_version ` or `installed_postgresql_package_source `. This way those who have more than one PG version on a machine can force one to be used. Also, I'm suggesting to implement this so that upgrades can be performed without having to write custom resource. Thanks for the feedback

Reply:
@bmhughes As @Justin-Fernbaugh mentioned, this logic is only needed during an in-place upgrade when you will have multiple versions installed. This provides a stable fix so that we can safely do this. This isn't intended to be used long term on a node.